store/src/java/com/zimbra/cs/imap/ImapHandler.java

```diff
@@ -1330,6 +1347,56 @@ boolean doLOGOUT(String tag) throws IOException {
         return false;
     }
 
+    private boolean checkZimbraAuth() {
+        return authenticator != null && authenticator.getMechanism().equals(ZimbraAuthenticator.MECHANISM);
+    }
+
+    boolean doFLUSHCACHE(String tag, CacheEntryType[] types, CacheEntry[] entries) throws IOException {
+        if (!checkState(tag, State.AUTHENTICATED)) {
+            return true;
+        } else if (!checkZimbraAuth()) {
+            sendNO(tag, "must be authenticated with X-ZIMBRA auth mechanism");
+            return true;
+        }
+        for (CacheEntryType type: types) {
+            try {
+                AuthToken authToken = ((ZimbraAuthenticator) authenticator).getAuthToken();
+                AdminAccessControl aac = AdminAccessControl.getAdminAccessControl(authToken);
+                Server localServer = Provisioning.getInstance().getLocalServer();
+                try {
+                    aac.checkRight(localServer, Admin.R_flushCache);
```

Comment:
nit: for efficiency reasons it might be nice to perform the static checkRight outside of the for loop.